### ISSUE TITLE
services.redis.image: image cannot have a tag and a digest at line 54 column 12

### ISSUE THREAD

**Issue by akostadinov** [containers/podlet#101]
Hello, it seems like podman supports specifying both - tag and digest at the same time. See containers/common#579

So I guess podlet should do too.

**Comment by k9withabone** [comment]
I'm guessing you are asking about this for a compose file. While Podman does support specifying both a tag and digest, the [Compose Specification](https://github.com/compose-spec/compose-spec/blob/231b09c30d339e950c0da17fe5bdc793366b8fde/05-services.md#image) shows that it must be one or the other, not both.

**Comment by k9withabone** [comment]
I'm not ignoring this. Did you read my reply? Did you see the link? The [Compose Specification](https://github.com/compose-spec/compose-spec) is primarily made by Docker for Docker and is what `docker compose` v2 is based on. If you think it should be changed with regard to specifying an image, you should create an issue there.

**Comment by akostadinov** [comment]
Ok, I thought it was related to podman. I don't think it has to be changed. 

I see that specifying both is accepted by docker though. Whether they intentionally or not write a spec they don't follow is a different topic. Example https://github.com/immich-app/immich/releases/latest/download/docker-compose.yml

I think it makes sense and is straightforward to support. It has usefulness I assume to specify both so that one can figure out what image is being tracked by the compose author.

**Comment by k9withabone** [comment]
Fair enough. I'll reopen this, but it will probably be a bit until it's resolved as it requires making breaking changes to my [`compose_spec`](https://github.com/k9withabone/compose_spec_rs) library and I just released a new major version last week.

**Comment by strongoose** [comment]
I hit this exact problem with Immich today - frustrating that docker don't seem to actually follow their own spec!
